apis/v1alpha1/upstreamsettingspolicy_types.go

```diff
@@ -0,0 +1,96 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	gatewayv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:categories=nginx-gateway-fabric,scope=Namespaced,shortName=uspolicy
+// +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+// +kubebuilder:metadata:labels="gateway.networking.k8s.io/policy=direct"
+
+// UpstreamSettingsPolicy is a Direct Attached Policy. It provides a way to configure the behavior of
+// the connection between NGINX and the upstream applications.
+type UpstreamSettingsPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of the UpstreamSettingsPolicy.
+	Spec UpstreamSettingsPolicySpec `json:"spec"`
+
+	// Status defines the state of the UpstreamSettingsPolicy.
+	Status gatewayv1alpha2.PolicyStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// UpstreamSettingsPolicyList contains a list of UpstreamSettingsPolicies.
+type UpstreamSettingsPolicyList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []UpstreamSettingsPolicy `json:"items"`
+}
+
+// UpstreamSettingsPolicySpec defines the desired state of the UpstreamSettingsPolicy.
+type UpstreamSettingsPolicySpec struct {
+	// ZoneSize is the size of the shared memory zone used by the upstream. This memory zone is used to share
+	// the upstream configuration between nginx worker processes. The more servers that an upstream has,
+	// the larger memory zone is required.
+	// Default: OSS: 512k, Plus: 1m.
+	// Directive: https://nginx.org/en/docs/http/ngx_http_upstream_module.html#zone
+	//
+	// +optional
+	ZoneSize *Size `json:"zoneSize,omitempty"`
+
+	// KeepAlive defines the keep-alive settings.
+	//
+	// +optional
+	KeepAlive *UpstreamKeepAlive `json:"keepAlive,omitempty"`
+
+	// TargetRefs identifies API object(s) to apply the policy to.
+	// Objects must be in the same namespace as the policy.
+	// Support: Service
+	//
+	// +kubebuilder:validation:MaxItems=16
+	// +kubebuilder:validation:XValidation:message="TargetRefs Kind must be: Service",rule="self.exists(t, t.kind=='Service')"
+	// +kubebuilder:validation:XValidation:message="TargetRefs Group must be gateway.networking.k8s.io.",rule="self.all(t, t.group=='gateway.networking.k8s.io')"
```

Comment:
A Service object is not a part of the gateway.networking group, it's a part of the core k8s group. The gateway.networking group contains all the Gateway API CRDs.

So group should allow for an empty string or "core". I think that's how we do it elsewhere, right @kate-osborn?

Also something I noticed with [BackendTLSPolicy](https://github.com/kubernetes-sigs/gateway-api/blob/main/apis/v1alpha3/backendtlspolicy_types.go) is that they have a min items for the targetRefs. We haven't done that with any of our policies yet, but maybe we should.

Reply:
> So group should allow for an empty string or "core". I think that's how we do it elsewhere, right @kate-osborn?
Yep, good catch.

> Also something I noticed with [BackendTLSPolicy](https://github.com/kubernetes-sigs/gateway-api/blob/main/apis/v1alpha3/backendtlspolicy_types.go) is that they have a min items for the targetRefs. We haven't done that with any of our policies yet, but maybe we should.

Yeah, we should. That's a miss on our part. Can we add that to ObservabilityPolicy without bumping the version? I think tightening validation is a breaking change....

Reply:
Good point about breaking...part of me thinks it's okay because if a user had 0 items, the policy wouldn't function. And adding this restriction wouldn't affect any functional policies.

Here's another thought...would we want to allow users to have an empty list to essentially "disable" the policy without deleting it? Is there value in that?

Reply:
> Good point about breaking...part of me thinks it's okay because if a user had 0 items, the policy wouldn't function. And adding this restriction wouldn't affect any functional policies.

True. We could probably get away with not bumping the version

> Here's another thought...would we want to allow users to have an empty list to essentially "disable" the policy without deleting it? Is there value in that?

I don't think there's much value in that. I could see a user wanting to move a policy from one resource to another -- say during an upgrade -- but I can't think of why a user would want to keep a policy around that has no effect.